### ISSUE TITLE
Setting number of cross-validation folds

### ISSUE THREAD

**Issue by m-petersen** [dorianps/LESYMAP#42]
Dear Dorian,

thank you for the great package!

I am trying to implement a cross-validation of lesion symptom mapping between lesion and cognitive data with lesymap. Given your comments in this issue (https://github.com/dorianps/LESYMAP/issues/33) I understand that lesymap already does a cross-validation and I can simply report "CVcorrelation.stat" as a measure of out-of-sample prediction performance, right? To achieve a 10-times repeated 10-fold cross-validation I can set cvRepetitions=10 and nFolds=10 as in `lsm = lesymap(filenames_attention, behavior_attention, method = 'sccan', optimizeSparseness=TRUE, validateSparseness=TRUE, correctByLesSize="voxel", minSubjectPerVoxel = 5, cvRepetitions=10, nFolds=10)`, correct? Just checking to be on the safe side. :)

Is it possible to extract the Pearson correlations for each of the 100 folds? I would like to plot the distribution.

Best,
Marvin

**Comment by dorianps** [comment]
Hi Marvin,

I should clarify something from the old thread. The internal cross-validation with SCCAN is done simply to select the correct sparseness value. Basically we search for optimal sparseness and the best way to find the optimal one is to run internal cross-validation and make sure the voxel weights obtained from the training set (e.g., 90% of subjects) can predict behavior in the test set (e.g., 10% of subjects). Once the sparseness is found and there is evidence of good predictive value from the Nfold procedure, a final sccan is run with the identified sparseness, and that is what you get as results. Strictly speaking, the final outcome is not tested with cross-validation, but because smaller train-test sample predictions worked, it is assumed that the map you get is similar to the maps recovered during the cross-validation procedure (and therefore at least as predictive as the average CV correlation across folds and repetitions). There is also that arbitrary threshold we use to zero out all voxels with weights less than 10% of maximal weight. The formal statistical map you get at the end does not have those voxels, but a prediction would have them. The intent at the time was to find ways to map functions, not to test thoroughly the behavioral prediction. So, I have not tested how prediction works without those low weighted voxels.

Yes, you can use the `cvRepetition` and `nFolds` you select, that is fine. The default for `nFolds` is 4 because 25% test gives more stable results than 10% test rate. On the other hand, `cvRepetitions` adapt to the sample size, smaller samples need more repetitions to stabilize the results, code is here: https://github.com/dorianps/LESYMAP/blob/master/R/optimize_SCCANsparseness.R#L52-L54

No, the individual correlations are not shown in results. This is the line where repetitions are averaged: https://github.com/dorianps/LESYMAP/blob/master/R/optimize_SCCANsparseness.R#L130. You would need to modify the code, the easiest way is to print or save `CVcorr` before it is averaged.

Another useful function may be `lesymap_predict()`. Basically it takes an final output and predicts new subjects, so you could basically run sccan with your predefined sparseness on training subjects using `lesymap()`  and then predict using `lesymap_predict()`. I only built the predict function for a lab that wanted to predict new subjects and I have not used it myself thoroughly (see papers by Aaron Boes).

**Comment by dorianps** [comment]
By the way, regarding this:

> Is it possible to extract the Pearson correlations for each of the 100 folds? I would like to plot the distribution.

You have 10 correlations from `cvRepetitions` not 100. If you try to run correlations on the miniscule 10% of data for each `nFold`, things may be unstable. There is too few subjects in a single fold, this is why we complete the full vector of predictions from all subjects before running a correlation.

**Comment by m-petersen** [comment]
Many thanks for your insights! This clarifies everything.
